fendermint/actors/blobs/src/state/blobs.rs

```diff
@@ -57,9 +59,136 @@ pub struct BlobsProgressCollection {
 /// and an Iroh node [`PublicKey`].
 type BlobSource = (Address, SubscriptionId, PublicKey);
 
-/// A set of [`BlobSource`]s.
+/// A set of [`BlobSource`]s implemented with Hamt.
 /// A blob in the collection may have multiple sources.
-type BlobSourceSet = HashSet<(Address, SubscriptionId, PublicKey)>;
+#[derive(Debug, Clone, Serialize_tuple, Deserialize_tuple, PartialEq)]
+pub struct BlobSourceSet {
+    pub root: hamt::Root<String, ()>,
+    size: u64,
+}
+
+fn convert_slice_to_array(slice: &[u8]) -> Result<[u8; 32], ActorError> {
```

Comment:
IMO it might be better to revamp it into something more sematically sound, like `pubkey_as_array`?